### PR TITLE
Refactoring stage II - move common functions from infinispan_controller to the separate packages

### PR DESCRIPTION
**Pull request by dmvolod**
Part of the #290 implementation
Stage II - move common functions from infinispan_controller to the separate packages. Arrange Status structs for Cluster and Cache (change from string to metav1.ConditionStatus in Cluster)

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla please have a look at this refactoring PR

**Comment by rigazilla** [comment]
@dmvolod can you rebase this? I can merge it before #489

**Comment by dmvolod** [comment]
Yes, sure, @rigazilla 
Will do this soon.

**Comment by dmvolod** [comment]
@rigazilla PR is ready, sorry a little bit huge, please review.

File: pkg/controller/utils/infinispan/configuration/xsite.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	restclient "k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateInfinispanConfiguration generates a server configuration
+func (xsite *XSite) CreateInfinispanConfiguration(name string, loggingCategories map[string]string, namespace string) InfinispanConfiguration {
```
Comment (rigazilla) [inline pkg/controller/utils/infinispan/configuration/xsite.go:22]: Is this func strictly related to xsite? It seems a general Infinispan function

File: pkg/controller/infinispan/infinispan_controller.go [hunk 2]
```diff
@@ -89,7 +87,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-// GetDiscoveryClient ...
+// TODO: If we will invoke getDiscoveryClient multiple times in one reconcile stage, it's necessary move 'discovery' to the ReconcileInfinispan struct
```
Comment (rigazilla) [inline pkg/controller/infinispan/infinispan_controller.go:90]: It would be good to add a short reason why

File: pkg/controller/infinispan/infinispan_controller.go [hunk 3]
```diff
@@ -431,6 +430,34 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		default:
 			reqLogger.Info("Cluster NOT exposed. Unsupported type?", "ExposeType", infinispan.Spec.Expose.Type)
 		}
+	} else if err != nil {
```
Comment (rigazilla) [inline pkg/controller/infinispan/infinispan_controller.go:433]: please check this snippet, seems to me a dup of lines 287+

File: test/e2e/constants/constants.go [hunk 4]
```diff
@@ -17,16 +17,15 @@ const (
 	DefaultPollPeriod = 1 * time.Second
 	// Maximum time to wait for resources
 	MaxWaitTimeout      = 120 * time.Second
-	DefaultClusterPort  = 11222
+	DefaultHotRodPort   = 11222
```
[the remark this answers is not in the document]
Comment (dmvolod) [reply test/e2e/constants/constants.go:20]: Sure, changed to DefaultClusterPort. But the port in SS is Name: "hotrod". Do we need the new generic name for this?

File: pkg/controller/utils/infinispan/configuration/xsite.go (hunk 1 above)
Comment (dmvolod) [reply pkg/controller/utils/infinispan/configuration/xsite.go:22]: Oh, yes, it was originally in configuration.go and suddenly moved, fixed

File: pkg/controller/infinispan/infinispan_controller.go (hunk 3 above)
Comment (dmvolod) [reply pkg/controller/infinispan/infinispan_controller.go:433]: Yes, thanks. This was regression from complex merge.

File: pkg/controller/infinispan/infinispan_controller.go (hunk 2 above)
Comment (dmvolod) [reply pkg/controller/infinispan/infinispan_controller.go:90]: yes, i will add that due to performance reason

**Comment by dmvolod** [comment]
@rigazilla thanks for review.

File: test/e2e/constants/constants.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rigazilla) [reply test/e2e/constants/constants.go]: Oh yeah we should change also that one, but I think we need a more descriptive name for that. 
`ClusterPort` or `SinglePort` seem too generic imo. Wdyt?

[the remark this answers is not in the document]
Comment (rigazilla) [reply test/e2e/constants/constants.go]: maybe we can change it to `ispn` ?

[the remark this answers is not in the document]
Comment (dmvolod) [reply test/e2e/constants/constants.go]: wdyt about InfinispanClusterPort?
But I saw an issue where asked abou separate HotRod/REST/Console to different ports. Is it make sense?

[the remark this answers is not in the document]
Comment (rigazilla) [reply test/e2e/constants/constants.go]: > wdyt about InfinispanClusterPort?

yep, if it's ok I would go for a shorter InfinispanPort and infinispan as SS containerPort name.

> But I saw an issue where asked abou separate HotRod/REST/Console to different ports. Is it make sense?

This needs some thoughts, if it's really needed I think it impacts also how the the server configuration (infinispan.xml) is generated.

[the remark this answers is not in the document]
Comment (dmvolod) [reply test/e2e/constants/constants.go]: > > wdyt about InfinispanClusterPort?
> 
> yep, if it's ok I would go for a shorter InfinispanPort and infinispan as SS containerPort name.
> 
Agree, fixed. Thanks, @rigazilla 

> > But I saw an issue where asked abou separate HotRod/REST/Console to different ports. Is it make sense?
> 
> This needs some thoughts, if it's really needed I think it impacts also how the the server configuration (infinispan.xml) is generated.
Yes, sure, looks like the current template and generator are not able to separate ports.

**Comment by rigazilla** [comment]
merged, thanks @dmvolod !
